honeybee_schema/radiance/material.py

```diff
@@ -0,0 +1,280 @@
+"""Material Schema"""
+from pydantic import Field, constr
+from typing import List, Union
+
+from .._base import IDdBaseModel
+
+
+class Void(IDdBaseModel):
+    """Void modifier"""
+
+    type: constr(regex='^Void$') = 'Void'
+
+
+class ModifierBase(IDdBaseModel):
+    """Base class for Radiance Modifiers"""
+
+    # TODO: We can't define union of schema objects that haven't been
+    # defined yet. We also can't move this down, because ModifierBase
+    # is inherited by all these objects.
+    modifier: Union[
+         Void, Plastic, Glass, BSDF, Glow, Light, Trans
+    ] = Field(
```

Comment:
Pydantic supports the concept of self referencing. We use it Queenbee for referencing nested workflows. Should also work here I guess. Here is the documentation: https://pydantic-docs.helpmanual.io/#self-referencing-models

Here is the Queenbee example which is slightly different but should be a good reference:

https://github.com/ladybug-tools/queenbee/blob/c4693b1b628a931d85c466513e6d3de85401c7cb/queenbee/schema/workflow.py#L46

https://github.com/ladybug-tools/queenbee/blob/c4693b1b628a931d85c466513e6d3de85401c7cb/queenbee/schema/workflow.py#L549

Reply:
Wow, Pydantic is awesome.

Reply:
Sweet, got it implemented.